### ISSUE TITLE
[Refactor] improve codebase structure & our team section

### ISSUE THREAD

**Issue by Anmol-Baranwal** [rupali-codes/LinksHub#2315]
### File Name

Targeting whole codebase

### Reason for Refactoring the Code

- Refactoring our team section (using `clsx` & `tw-merge`) to write a better code.
- Creating a new Icons system so that we can use SVG icons with ease -> improves codebase structure. (using `lucide-icons`)
   - We won't have to use the small bundle in different components and we can just import at one place.
- Add priority loading to all the images and use the recent `next/legacy/image` components.
- fix all linting errors

@rupali-codes 
Should I add a classnames order plugin so it will automatically arrange the tailwind classnames in the correct order.

I want to do this myself. Which of the points do you think is relevant to the current codebase? Let me know and I will cover all of them in a single PR.

### Checklist

- [X] I have checked the existing [issues](https://github.com/rupali-codes/LinksHub/issues).
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md).
- [X] The changes don't break the code and don't introduce new functionality.
- [X] I am willing to work on this issue (optional).

**Comment by rupali-codes** [comment]
@Anmol-Baranwal Yes, it would make the code more clear. And please let me know if you need any help, I'll contribute to the same PR :)

**Comment by Anmol-Baranwal** [comment]
@rupali-codes, 
I've started working on this. It seems that `lucide-icons` won't work because many icons in the current codebase are from different categories even under React icons like fa, fa6, io.

I've created a component for React icons itself and just need to cross-check that all the icons work like the original.

Regarding the maintainers section code, it's good enough (just realized it). While we could use `clsx` and `tw-merge`, it would add more complexity than the current setup, so I think it's better not to replace it. I did little changes and current one seems better.

Everything else is done. I'll make a PR in 1-2 days.
